usmqe/web/tendrl/task_wait.py

```diff
@@ -9,75 +9,107 @@
 import datetime
 import pytest
 
+from usmqe.web.tendrl.mainpage.tasks.pages import TaskDetails, TaskEvents
+
 
 class TaskWaitException(Exception):
     """
     task wait exception
     """
 
 
-def task_wait(task_details, desired_state='Finished', ttl=600):
+def task_wait(driver, desired_state='Finished', ttl=600,
+              update_time=600, sleep_time=5):
     """
     wait till the task has desired state
+    NOTE: it has to be on the page with task details
 
     Parameters:
-        task_details: webstr page object with task details
+        driver: selenium driver
         desired_state (str): task desired state
                              'New' -> 'Processing' -> 'Finished'/'Failed'
                              where 'New' is not accepted as desired state
         ttl (int): how long it waits till the tasks is finished
                    in seconds
+        update_time (int): how long it waits till the new message (event)
+                            is comming
+                           it's also the time how long the task could
+                            remain in 'New' state
+                           in seconds
+        sleep_time (int): sleep time between loops
     """
+    task_details = TaskDetails(driver)
+
     # Wait till the task has the correct state
     # state of any task: New -> Processing -> Finished/Failed
     if desired_state == 'New':
         # initial task state, no need to wait for it
         return
```

Comment:
Would it make more sense to raise exception "waiting for new doesn't make sense"?